plus/src/test/java/org/apache/calcite/slt/SqlLogicTestsForCalciteTests.java

```diff
@@ -0,0 +1,312 @@
+     * Check if 'other' summaries have regressions compared to `this`.
+     *
+     * @return 'true' if other contains regressions.
+     * @param other  Test results to compare with.
+     *               'other' can contain only a subset of the tests.
+     */
+    boolean regression(AllTestSummaries other) {
+      boolean regression = false;
+      for (TestSummary summary: other.testResults.values()) {
+        TestSummary original = this.testResults.get(summary.file);
+        if (original == null) {
+          System.err.println("No historical data for test " + summary.file);
+          continue;
+        }
+        if (original.regression(summary)) {
+          System.err.println("Regression: " + original.file
+              + " had " + original.failed + " failures, now has " + summary.failed);
+          regression = true;
+        }
+      }
+      return regression;
+    }
+
+    @Override public String toString() {
+      List<TestSummary> results = new ArrayList<>(this.testResults.values());
+      results.sort(Comparator.comparing(left -> left.file));
+      StringBuilder result = new StringBuilder();
+      for (TestSummary summary: results) {
+        result.append(summary.toString());
+        result.append(System.lineSeparator());
+      }
+      return result.toString();
+    }
+
+    /**
+     * Write the test results to the specified file.
+     */
+    public void writeToFile(File file) throws IOException {
+      try (BufferedWriter writer =
+               new BufferedWriter(
+                   new OutputStreamWriter(
+                       Files.newOutputStream(file.toPath()), StandardCharsets.UTF_8))) {
+        writer.write(this.toString());
+      }
+    }
+
+    /**
+     * True if there is an entry for the specified test file.
+     *
+     * @param test Test file name.
+     */
+    public boolean contains(String test) {
+      return this.testResults.containsKey(test);
+    }
+  }
+
+  /**
+   * Summaries produced for the current run.
+   * Must be static since it is written by the `findRegressions`
+   * static method.
+   */
+  static AllTestSummaries testSummaries = new AllTestSummaries();
+  /**
+   * Summaries checked-in as resources that we compare against.
+   */
+  static AllTestSummaries goldenTestSummaries = new AllTestSummaries();
+
+  static final String GOLDENFILE = "/slttestfailures.txt";
+
+  private static TestStatistics launchSqlLogicTest(String... args) throws IOException {
+    OptionsParser options = new OptionsParser(false, System.out, System.err);
+    CalciteExecutor.register(options);
+    return net.hydromatic.sqllogictest.Main.execute(options, args);
+  }
+
+  TestSummary shortSummary(String file, TestStatistics statistics) {
+    return new TestSummary(file, statistics.getPassedTestCount(), statistics.getFailedTestCount());
+  }
+
+  // The following tests currently timeout during execution.
+  // Technically these are Calcite bugs.
+  Set<String> timeout = new HashSet<String>() {
```

Comment:
Not worth sub-classing `HashSet` for this. Use `ImmutableSet.of` or `ImmutableSet.builder`. Same in other places where this pattern is used.